test/dotnet.Tests/GivenThatTheUserIsRunningDotNetForTheFirstTime.cs

```diff
@@ -28,6 +28,7 @@ static GivenThatTheUserIsRunningDotNetForTheFirstTime()
                 .WithWorkingDirectory(testDirectory.Path);
             command.Environment["NUGET_PACKAGES"] = testNugetCache;
             command.Environment["DOTNET_SKIP_FIRST_TIME_EXPERIENCE"] = "";
+            command.Environment["SkipInvalidConfigurations"] = "true";
```

Comment:
The problem is that we're calling restore within the code under test to hydrate the nuget cache. That code dotnet-new's and then restores. I had to move it to restore3 since the dotnet-new template now uses csproj instead of project.json.